card.go

```diff
@@ -1,49 +1,105 @@
 package goker
 
+import "fmt"
+
 type suit int8
-type rank int8
 
 const (
+	// Spade - Constant representing the suit ♠
 	Spade suit = iota
+	// Heart - Constant representing the suit ♥
 	Heart
+	// Diamond - Constant representing the suit ♦
 	Diamond
+	// Club - Constant representing the suit ♣
 	Club
 )
 
+type rank int8
+
 const (
+	// Two - Constant representing the rank 2
 	Two rank = iota + 2
+	// Three - Constant representing the rank 3
 	Three
+	// Four - Constant representing the rank 4
 	Four
+	// Five - Constant representing the rank 5
 	Five
+	// Six - Constant representing the rank 6
 	Six
+	// Seven - Constant representing the rank 7
 	Seven
+	// Eight - Constant representing the rank 8
 	Eight
+	// Nine - Constant representing the rank 9
 	Nine
+	// Ten - Constant representing the rank 10
 	Ten
+	// Jack - Constant representing the rank Jack
 	Jack
+	// Queen - Constant representing the rank Queen
 	Queen
+	// King - Constant representing the rank King
 	King
+	// Ace - Constant representing the rank Ace
 	Ace
 )
 
-type PlayingCard interface {
+// Card - Represents a playing card, with a suit and rank
+type Card interface {
 	Rank() rank
 	Suit() suit
+	String() string
 }
 
 type card struct {
 	rank rank
 	suit suit
 }
 
-func (c card) Rank() rank {
+func (c *card) Rank() rank {
 	return c.rank
 }
 
-func (c card) Suit() suit {
+func (c *card) Suit() suit {
 	return c.suit
 }
 
-func NewCard(r rank, s suit) card {
-	return card{r, s}
+func (c *card) String() string {
+	var suitStr string
+	switch c.Suit() {
+	case Spade:
+		suitStr = "♠"
+	case Heart:
+		suitStr = "♥"
+	case Diamond:
+		suitStr = "♦"
+	case Club:
+		suitStr = "♣"
+	}
+
+	var rankStr string
+	switch c.Rank() {
+	case Ace:
+		rankStr = "A"
+	case King:
+		rankStr = "K"
+	case Queen:
+		rankStr = "Q"
+	case Jack:
+		rankStr = "J"
+	case Ten:
+		rankStr = "T"
+	default:
+		rankStr = fmt.Sprintf("%d", int(c.Rank()))
+	}
+
+	return rankStr + suitStr
+}
+
+// NewCard - Constructs a new card with the given suit and rank
+func NewCard(r rank, s suit) Card {
```

Comment:
Considering the above comment about interfaces vs. structs, I read this at first as a typo of `func NewCard(r rank, s suit) *Card {` until I realized `Card` is an interface (*especially* since this is returning `&c`).

The `NewThing()` naming idiom is typically for constructors for initializing any struct type with nice, empty default values.  We could also omit the constructor entirely and do something like:
 
https://golang.org/doc/effective_go.html#composite_literals

Reply:
> The NewThing() naming idiom is typically for constructors for initializing a type with nice, empty default values and then returning a pointer.

Oh, hmm. I've mostly been using it like an `init` in Swift, to ensure that all expected fields are set when the struct is first created. When you say "nice, empty default values," do you mean the Go idiom would be not

```
func newStraightFlush(highCard rank) *straightFlush {
	sf := straightFlush{highCard}
	return &sf
 }

sf := newStraightFlush(Ace)
```

but instead

```
func newStraightFlush() *straightFlush {
	sf := straightFlush{}
	return &sf
 }

sf := newStraightFlush()
sf.rank = Ace
```
?

In any case, I hear you re: not using interface as a return type.